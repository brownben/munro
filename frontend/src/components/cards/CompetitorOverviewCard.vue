<template>
  <section
    class="flex flex-col items-center justify-around col-span-2 px-4 py-5 text-center bg-white shadow-md lg:col-span-1 rounded-shape-xl sm:px-6 sm:flex-row sm:justify-between sm:text-left lg:px-6"
  >
    <div class="flex-shrink text-center sm:text-left">
      <router-link
        :to="`/leagues/${competitor.league}`"
        class="text-xs font-bold leading-tight tracking-widest uppercase font-heading text-main-700"
      >
        {{ competitor.league }}
      </router-link>
      <h2
        class="mt-1 mb-2 text-3xl font-bold leading-tight tracking-tight text-gray-900 font-heading sm:tracking-normal"
      >
        {{ competitor.name }}
      </h2>
      <p class="text-lg leading-tight text-gray-500 font-heading">
        {{ competitor.course }}
        <span
          v-if="competitor.course && (competitor.club || competitor.ageClass)"
          class="mx-1"
        >
          -
        </span>
        {{ competitor.ageClass }}
        <span v-if="competitor.ageClass && competitor.club" class="mx-1">
          -
        </span>
        {{ competitor.club }}
      </p>
    </div>

    <router-link
      class="inline-block w-auto px-3 py-2 mt-3 text-base leading-tight whitespace-no-wrap transition duration-300 ease-in-out select-none text-main-700 font-heading rounded-shape hover:text-main-800 focus:text-main-800 sm:my-0 hover:bg-main-100 focus:bg-main-100 sm:mr-0 sm:ml-4"
      :to="`/competitors/${competitor.id}`"
      >View Results â†’</router-link
    >
  </section>
</template>

<script>
export default {
  props: {
    competitor: { type: Object, default: () => ({}) },
  },
}
</script>
