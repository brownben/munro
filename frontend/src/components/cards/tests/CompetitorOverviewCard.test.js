import { shallowMount } from '@vue/test-utils'
import CompetitorOverviewCard from '@/components/cards/CompetitorOverviewCard.vue'

import axios from 'axios'

jest.mock('axios')

beforeEach(() => {
  jest.resetAllMocks()
  axios.get.mockRejectedValue()
})

test('Is a Vue Instance', () => {
  const wrapper = shallowMount(CompetitorOverviewCard, {
    stubs: ['router-link'],
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = shallowMount(CompetitorOverviewCard, {
    propsData: {
      competitor: {
        name: 'Bob Jones',
        course: 'Red',
        ageClass: 'M40',
        club: 'HAT',
        id: 7,
      },
    },
    stubs: ['router-link'],
  })
  expect(wrapper.element).toMatchSnapshot()
})
