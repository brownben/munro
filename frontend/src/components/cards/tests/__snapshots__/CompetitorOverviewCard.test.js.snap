// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<section
  class="flex flex-col items-center justify-around col-span-2 px-4 py-5 text-center bg-white shadow-md lg:col-span-1 rounded-shape-xl sm:px-6 sm:flex-row sm:justify-between sm:text-left lg:px-6"
>
  <div
    class="flex-shrink text-center sm:text-left"
  >
    <router-link-stub
      class="text-xs font-bold leading-tight tracking-widest uppercase font-heading text-main-700"
      to="/leagues/undefined"
    >
      
      
    
    </router-link-stub>
     
    <h2
      class="mt-1 mb-2 text-3xl font-bold leading-tight tracking-tight text-gray-900 font-heading sm:tracking-normal"
    >
      
      Bob Jones
    
    </h2>
     
    <p
      class="text-lg leading-tight text-gray-500 font-heading"
    >
      
      Red
      
      <span
        class="mx-1"
      >
        
        -
      
      </span>
      
      M40
      
      <span
        class="mx-1"
      >
        
        -
      
      </span>
      
      HAT
    
    </p>
  </div>
   
  <router-link-stub
    class="inline-block w-auto px-3 py-2 mt-3 text-base leading-tight whitespace-no-wrap transition duration-300 ease-in-out select-none text-main-700 font-heading rounded-shape hover:text-main-800 focus:text-main-800 sm:my-0 hover:bg-main-100 focus:bg-main-100 sm:mr-0 sm:ml-4"
    to="/competitors/7"
  >
    View Results â†’
  </router-link-stub>
</section>
`;
